### DISCUSSION TITLE
How to detect what connections a software / application is trying to make

### DISCUSSION THREAD

**Discussion by itsnotsaved** [AdguardTeam/AdGuardHome#3276] [Q&A] [answered]
I have AdGuardHome installed and multiple devices connected to it. One device try to keep connect to specific domain continuously that already blocked but i dont know how to detect which application is trying for connect to that domain. Basically i tried to detect by closing applications and ` netstat` tool doesn't help. (In debian based distro)

**Comment by ameshkov** [comment, marked as answer]
Not an easy thing to do considering that the app probably uses `getaddrinfo` and the DNS query is sent by the system.

You probably need to:
1. Unblock that domain or use `$dnsrewrite` to point it to some real IP address
2. Use `netstat` and look for connections that that IP address to find the process
